crates/storage-proto/proto/dev/restate/storage/v1/domain.proto

```diff
@@ -14,6 +14,13 @@ message FullInvocationId {
     bytes invocation_uuid = 3;
 }
 
+message MaybeFullInvocationId {
+    oneof kind {
+        FullInvocationId full_invocation_id = 1;
+        bytes invocation_id = 2;
+    }
```

Comment:
Premature optimization: Rather than this representation with nested messages, perhaps we could just do it as follows:

```protobuf
message MaybeFullInvocationId {
  int64 partition_key = 1;
  bytes invocation_uuid = 2;

  // If service name missing, this is InvocationId, otherwise it's FullInvocationId
  bytes service_name = 3;
  bytes service_key = 4;
}
```

Flattening the protobuf representation is generally more efficient than many nested messages.

Reply:
Indeed this is maybe premature optimizations since I am not sure whether we enforce that service names can not be empty atm. And even if we do (which probably makes sense), it introduces a subtle coupling that might break once we change things. Not sure whether the performance gains justify that.